### PR TITLE
[Sign/Web3Wallet] Add pending proposals

### PR DESCRIPTION
**Pull request by alexander-lsvk**
## Due Dilligence

* [ ] Breaking change
* [x] Requires a documentation update

### CODE REVIEW

File: Sources/WalletConnectSign/Services/HistoryService.swift [hunk 1]
```diff
@@ -39,4 +67,25 @@ private extension HistoryService {
...
+        guard let proposal = try? record.request.params?.get(SessionType.ProposeParams.self)
+        else { return nil }
+        return proposal
+    }
+    
+    func mapProposalRecord(_ record: RPCHistory.Record) -> Session.Proposal? {
+        guard let proposal = try? record.request.params?.get(SessionType.ProposeParams.self)
+        else { return nil }
+        
+        return Session.Proposal(
```
Comment (llbartekll) [inline Sources/WalletConnectSign/Services/HistoryService.swift:81]: you can use SessionProposal's `publicRepresentation(pairingTopic: String) -> Session.Proposal` here

File: Sources/WalletConnectSign/Services/HistoryService.swift [hunk 2]
```diff
@@ -17,15 +27,33 @@ final class HistoryService {
...
+                guard let proposalParams = mapProposeParams(record) else {
+                    return nil
+                }
+                return RequestSubscriptionPayload(id: record.id, topic: record.topic, request: proposalParams, publishedAt: Date())
+            }
+        
+        requestSubscriptionPayloads.forEach {
+            let proposal = $0.request
+            proposalPayloadsStore.set($0, forKey: proposal.proposer.publicKey)
+        }
```
Comment (llbartekll) [inline Sources/WalletConnectSign/Services/HistoryService.swift:45]: you do it to set values to proposalPayloadsStore on app launch?

File: Sources/WalletConnectSign/Services/HistoryService.swift [hunk 3]
```diff
@@ -17,15 +27,33 @@ final class HistoryService {
...
+    
+    func getPendingProposals() -> [Session.Proposal] {
+        let pendingHistory = history.getPending()
+        
+        let requestSubscriptionPayloads = pendingHistory
+            .compactMap { record -> RequestSubscriptionPayload<SessionType.ProposeParams>? in
+                guard let proposalParams = mapProposeParams(record) else {
+                    return nil
+                }
+                return RequestSubscriptionPayload(id: record.id, topic: record.topic, request: proposalParams, publishedAt: Date())
```
Comment (llbartekll) [inline Sources/WalletConnectSign/Services/HistoryService.swift:39]: I don't like we are recreating RequestSubscriptionPayload type here.
could we change type stored by proposalPayloadsStore?

File: Sources/WalletConnectSign/Services/HistoryService.swift (hunk 2 above)
Comment (llbartekll) [reply Sources/WalletConnectSign/Services/HistoryService.swift:45]: make sense

File: Sources/WalletConnectSign/Services/HistoryService.swift [hunk 4]
```diff
@@ -17,15 +27,33 @@ final class HistoryService {
...
+            }
+        
+        requestSubscriptionPayloads.forEach {
+            let proposal = $0.request
+            proposalPayloadsStore.set($0, forKey: proposal.proposer.publicKey)
```
Comment (flypaper0) [inline Sources/WalletConnectSign/Services/HistoryService.swift:44]: why we cannot remove proposalPayloadsStore and depend on rpc history only? 

If big array of expired proposals the only reason I think it's fine. We can fix it when implement expiration logic for proposals

**Pull request closed without merging**